### DISCUSSION TITLE
Firebase/Android push, OnEntry not working

### DISCUSSION THREAD

**Discussion by Yahya0007** [shinyorg/shiny#1385] [Q&A]
Hi

First time using Shiny 3 and very new at push notifications.

This is on Android. I am receiving notifications in OnReceived but no automatic pop up is coming up. This means there is nothing for user to click and OnEntry does not get called.

I am using this code to send message;

            var message = new Message()
            {
                Data = new Dictionary<string, string>
                {
                    { "click_action", "SHINY_PUSH_NOTIFICATION_CLICK" },
                    // Include additional data if needed for silent notifications
                    { "data", "my_silent_data" },
                    { "StaffID", "1001" },
                },
                Token = registrationToken,
                //Topic = "all",
                Notification = new Notification()
                {
                    Title = "Test from code",
                    Body = "Here is your test!"
                }
            };

My questions are;

1. How can I get OnEntry to fit in and get called?

2. How does Shiny notifications code work when app is in background or is not running?

Thanks

Regards

**Comment by aritchie** [comment]
> First time using Shiny 3 and very new at push notifications.

I understand that, but please note this isn't meant as tech support.  I give away a bunch of code and it is all open for you to view.  I've documented all of the client side for this.  I don't cover server side stuff because it really depends on the libraries you're using, but the click_action is part of the notification object for android and this is documented in the Android docs.

**Reply by Yahya0007** [reply]
Allan, thanks for your time. Appreciated.
